tests/pytests/integration/modules/test_mac_softwareupdate.py

```diff
@@ -0,0 +1,176 @@
+"""
+integration tests for mac_softwareupdate
+"""
+
+import pytest
+
+pytestmark = [
+    pytest.mark.slow_test,
+    pytest.mark.skip_if_binaries_missing("softwareupdate"),
+    pytest.mark.skip_if_not_root,
+    pytest.mark.skip_unless_on_darwin,
+    pytest.mark.skip_initial_gh_actions_failure,
+]
+
+
+@pytest.fixture(scope="function")
+def setup_teardown_vars(salt_call_cli):
+    ret = salt_call_cli.run("softwareupdate.list_ignored")
+    IGNORED_LIST = ret.data
+
+    ret = salt_call_cli.run("softwareupdate.schedule")
+    SCHEDULE = ret.data
+
+    ret = salt_call_cli.run("softwareupdate.get_catalog")
+    CATALOG = ret.data
+
+    try:
+        yield IGNORED_LIST, SCHEDULE, CATALOG
+    finally:
+        if IGNORED_LIST:
+            for item in IGNORED_LIST:
+                salt_call_cli.run_function("softwareupdate.ignore", item)
+        else:
+            salt_call_cli.run_function("softwareupdate.reset_ignored")
+
+        salt_call_cli.run_function("softwareupdate.schedule", SCHEDULE)
+
+        if CATALOG == "Default":
+            salt_call_cli.run_function("softwareupdate.reset_catalog")
+        else:
+            salt_call_cli.run_function("softwareupdate.set_catalog", CATALOG)
+
+
+def test_list_available(salt_call_cli, setup_teardown_vars):
+    """
+    Test softwareupdate.list_available
+    """
+    # Can't predict what will be returned, so can only test that the return
+    # is the correct type, dict
+    ret = salt_call_cli.run("softwareupdate.list_available")
+    assert isinstance(ret.data, dict)
```

Comment:
```suggestion
def test_list_available(salt_call_cli):
    """
    Test softwareupdate.list_available
    """
    # Can't predict what will be returned, so can only test that the return
    # is the correct type, dict
    ret = salt_call_cli.run("softwareupdate.list_available")
    assert isinstance(ret.data, dict)
```

Same for remaining tests